src/wallet/rpcwallet.cpp

```diff
@@ -126,6 +130,13 @@ UniValue getnewaddress(const UniValue& params, bool fHelp)
 
     pwalletMain->SetAddressBook(keyID, strAccount, "receive");
 
+    if (GetBoolArg("-usehd", true) && showDetails && pwalletMain->mapKeyMetadata[keyID].keypath.size() > 0)
+    {
+        UniValue result(UniValue::VOBJ);
+        result.pushKV("address", CBitcoinAddress(keyID).ToString());
+        result.pushKV("keypath", pwalletMain->mapKeyMetadata[keyID].keypath);
+        return result;
+    }
```

Comment:
showDetails ought to return Object format even when not generating from a HD key IMO.